### PR TITLE
RTCM base station proxy.

### PR DESCRIPTION
**Pull request by mookerji**
/cc @mfine @ljbade

### CODE REVIEW

**Comment by ljbade** [comment]
Looks good to me, but im no Haskell expert.

File: haskell/main/RTCM32RTCM3.hs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE LambdaCase        #-}
```
Comment (mfine) [inline haskell/main/RTCM32RTCM3.hs]: Don't think you're using `LambdaCase`

File: haskell/main/RTCM32RTCM3.hs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+{-# LANGUAGE RecordWildCards   #-}
```
Comment (mfine) [inline haskell/main/RTCM32RTCM3.hs]: Don't think you're using `RecordWildCards`

Comment (mfine) [inline haskell/main/RTCM32RTCM3.hs]: Oh, you are. Sorry.

File: haskell/main/RTCM32RTCM3.hs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import           Data.Conduit
+import           Data.Conduit.Binary
+import qualified Data.Conduit.List as CL
+import           Data.Conduit.Serialization.Binary
+import           Data.RTCM3
+import           Options.Generic
+import           System.IO
+
+data Args = Args
+  { newPosX   :: Maybe Double
```
Comment (mfine) [inline haskell/main/RTCM32RTCM3.hs]: You can remove the `Maybe` here to make these mandatory - then could remove all the below Maybe's as well.

File: haskell/main/RTCM32RTCM3.hs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+replace :: Args -> RTCM3Msg -> RTCM3Msg
+replace args (RTCM3Msg1005 m _rtcm3) = RTCM3Msg1005 new $ toRTCM3 new
+  where
+    new = newMsg1005 (argsToPos args) m
+replace args (RTCM3Msg1006 m _rtcm3) = RTCM3Msg1006 new $ toRTCM3 new
+  where
+    new = newMsg1006 (argsToPos args) m
+replace _ _rtcm3Msg                  = _rtcm3Msg
+
```
Comment (mfine) [inline haskell/main/RTCM32RTCM3.hs]: You could use the `LambdaCase` here.

File: haskell/main/RTCM32RTCM3.hs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+data Args = Args
+  { newPosX   :: Maybe Double
+  , newPosY   :: Maybe Double
+  , newPosZ   :: Maybe Double
+  } deriving ( Eq, Show, Generic )
+
+instance ParseRecord Args
+
+type PositionECEF = [Double]
```
Comment (mfine) [inline haskell/main/RTCM32RTCM3.hs]: The array and subsequent `!!` access is a little yuck - could just have a record, `Ecef`, and use it as the `Args` as well.

**Comment by mookerji** [comment]
Whoops, missed your review, @mfine. I'll update it later today or something.

**Comment by mfine** [comment]
Abandoned.

**Pull request closed without merging**